### PR TITLE
Disable revolutionary gamemode

### PR DESCRIPTION
**Pull request by mystery3525**
## About The Pull Request

Disables the Provocateur latejoin and Revolution roundstart dynamic rules, and sets the weight for the revolution gamemode to zero, effectively disabling it.

## Why It's Good For The Game

Typical revolutions devolve into the round ending within 20-30 minutes, or the round being dragged to 90+ minutes of nothing but trying to find the last head of staff or last head revolutionary being hidden or fuck off in space. This isn't an outright removal, but this just disables them unless someone can make this antagonist a little more fun/engaging.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Idk, check files? Also xenoarch is causing build errors.

</details>

## Changelog
:cl:
config: Disables the revolutionary gamemode and revolutionary dynamic ruleset
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
Finally someone did it, god bless.

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm [hunk 1]
```diff
@@ -99,7 +99,7 @@
 	enemy_roles = list(JOB_NAME_AI, JOB_NAME_CYBORG, JOB_NAME_SECURITYOFFICER,JOB_NAME_DETECTIVE,JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN, JOB_NAME_WARDEN)
 	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
 	required_candidates = 1
-	weight = 2
+	weight = 0
```
Comment (PowerfulBacon) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm]: Update these via the dynamic json

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by FriendlyContractor** [comment]
While i agree with round stalling being a problem i find myself greatly enjoying the gamemode. Are there really so few players enjoying revolution that its removal is warranted ?

**Comment by CydiaLamiales** [comment]
> Are there really so few players enjoying revolution that its removal is warranted ?

I'd say yes. Most of the time it just turns into a NRP murderfest.

**Comment by Rukofamicom** [comment]
> i find myself greatly enjoying gamemode majority of the time.

There's more to the equation than just being fun - there's a lot of things I enjoy that aren't a good fit for the server or conducive to an RP environment. I do mechanically enjoy team deathmatch and even FFA deathmatch (I especially love battle royale), But these aren't conducive to an RP setting at all. 

While it's good to factor how fun something is or can be into the equation, it needs to be weighed against the goals of the server and what sort of experience it wants to serve its players.

**Comment by PowerfulBacon** [comment]
I'll sign off on this :+1:

**Comment by itsmeow** [comment]
does the game start with this config though

**Comment by PowerfulBacon** [comment]
Wdym? We have to update the config on the server when we merge it

**Comment by itsmeow** [comment]
just wondering if weight:0 will cause issues with dynamic

**Comment by PowerfulBacon** [comment]
We should add a boolean disabled value to this config

**Comment by mystery3525** [comment]
So is there any changes I need to do?

**Comment by PowerfulBacon** [comment]
Add in a way to disable gamemodes properly to dynamic config

**Comment by itsmeow** [comment]
The official way to disable gamemodes is setting cost to 110. Just remove weight: 0

The cost 110 thing is intended and already used for meteors.

**Comment by mystery3525** [comment]
👍

**Review by itsmeow — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
That is incredibly dumb and shouldn't be the official way to remove gamemodes

**Comment by itsmeow** [comment]
out of scope

**Comment by itsmeow** [comment]
Soul removed

**Comment by itsmeow** [comment]
Config change applied

**Pull request merged**